daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala

```diff
@@ -383,37 +383,37 @@ class TransactionCoderSpec
       }
     }
 
-    // FIXME: https://github.com/digital-asset/daml/issues/7139
-    // enable the test when we have more that one version newer than 10
-    "fail if try to decode a node in a version newer than the enclosing Transaction message version" ignore {
+    "fail if try to decode a node in a version newer than the enclosing Transaction message version" in {
 
-      forAll(
-        danglingRefGenNode,
-        transactionVersionGen().filter(_ != V10),
-        transactionVersionGen().filter(_ != V10),
-        minSuccessful(10)) {
-        case ((nodeId, node), version1, version2) =>
-          whenever(version1 != version2) {
-            val (txVersion, nodeVersion) = inIncreasingOrder(version1, version2)
+      val postV10versions = TransactionVersion.All.filter(_ > V10)
 
-            val normalizedNode = minimalistNode(nodeVersion)(node).updateVersion(nodeVersion)
+      def twoDifferentVersions =
+        for {
+          v1 <- Gen.oneOf(postV10versions)
+          v2 <- Gen.oneOf(postV10versions.filter(_ != v1))
```

Comment:
How about
```suggestion
          v2 <- Gen.oneOf(postV10versions.filter(_ > v1))
```
and remove the call to `inIncreasingOrder` below? This seems to express the intent more clearly.

Reply:
Good idea.